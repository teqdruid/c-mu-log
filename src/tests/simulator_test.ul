/*this is test of simulator, it output the walls and trace of agent into agent1.dat*/

/*   ENV CODE... PLAYER---DON'T CHANGE OR LOOK AT ME!!!*/
goal(4,4);
size(20,20);
wall(12,4);
wall(4,9);
wall(6,8);
wall($X,$Y) {AND:
	$X>=4;
	$X<=6;	
	$Y<=15;
	$Y>=10;
	}


/* PLAYER CODE    */

imove("UP");
imove("UP");
imove("RIGHT");
imove("RIGHT");
imove("RIGHT");
imove("RIGHT");

move($dir) {
    @forget1( imove($dir); );
    imove($dir);
}

move("DOWN");


