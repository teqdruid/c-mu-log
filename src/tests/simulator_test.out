x: Moving UP

==== Turn 1 ====
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | . . . . . . . . . . . . . . . . 
. . . . . | . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . # . . . . . . . | . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
x . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 

x: Moving UP

==== Turn 2 ====
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | . . . . . . . . . . . . . . . . 
. . . . . | . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . # . . . . . . . | . . . . . . . . 
x . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 

x: Moving RIGHT

==== Turn 3 ====
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | . . . . . . . . . . . . . . . . 
. . . . . | . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . # . . . . . . . | . . . . . . . . 
. x . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 

x: Moving RIGHT

==== Turn 4 ====
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | . . . . . . . . . . . . . . . . 
. . . . . | . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . # . . . . . . . | . . . . . . . . 
. . x . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 

x: Moving RIGHT

==== Turn 5 ====
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | . . . . . . . . . . . . . . . . 
. . . . . | . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . # . . . . . . . | . . . . . . . . 
. . . x . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 

x: Moving RIGHT

==== Turn 6 ====
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | . . . . . . . . . . . . . . . . 
. . . . . | . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . # . . . . . . . | . . . . . . . . 
. . . . x . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 

x: Moving DOWN

==== Turn 7 ====
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | . . . . . . . . . . . . . . . . 
. . . . . | . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . # . . . . . . . | . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . x . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 

x: Moving DOWN

==== Turn 8 ====
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | | | . . . . . . . . . . . . . . 
. . . | . . . . . . . . . . . . . . . . 
. . . . . | . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . # . . . . . . . | . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . . . . . . . . . . . . . . . . . 
. . . . x . . . . . . . . . . . . . . . 

x: Moving DOWN

Simulation over: x hits the x margin and Game over!!! 

